exceptions/does_not_exist.py

```diff
@@ -0,0 +1,331 @@
+"""Custom exceptions for the exceptions where things do not exist."""
+
+from collections.abc import Sequence
+
+__all__: Sequence[str] = (
+    "RulesChannelDoesNotExistError",
+    "GuildDoesNotExistError",
+    "RoleDoesNotExistError",
+    "CommitteeRoleDoesNotExistError",
+    "GuestRoleDoesNotExistError",
+    "MemberRoleDoesNotExistError",
+    "ArchivistRoleDoesNotExistError",
+    "ChannelDoesNotExistError",
+    "RolesChannelDoesNotExistError",
+    "GeneralChannelDoesNotExistError",
+)
+
+
+import abc
+from typing import Final
+
+from classproperties import classproperty
+
+from exceptions import BaseDoesNotExistError, BaseTeXBotError
```

Comment:
Use relative imports for importing base exception classes. I.e. `from .base import ...` rather than ~`from exceptions import ...`~ or ~`from exceptions.base import ...`~